### DISCUSSION TITLE
How to add elixir bin and few others

### DISCUSSION THREAD

**Discussion by philippneugebauer** [coder/code-server#4856] [Q&A]
Hey guys,

that project looks awesome to me and seems like my perfect opportunity for a bit coding while travelling without laptop.

I set up my code server with docker but I am a bit confused how I am supposed to install the binary for erlang + elixir.

Just install it within the container?

And another small question. If some external content like extension descriptions are included, it's always stated that it couldn't connect. Is there anything that's required within the setup? I am using docker-compose with a nginx and normal websites work properly like the editor itself.

How do I use the ssh keys added within the machine? It works on the terminal of VS Code but I cannot push any changes within the git UI of VS Code.

**Comment by lg-nineoneninethreenine** [comment]
> I set up my code server with docker but I am a bit confused how I am supposed to install the binary for erlang + elixir.

code-server's base OS image is Debian. I don't know erlang and elixir, but have you tried getting curl, wget and some extra packages and try installing it classically on Linux?

> If some external content like extension descriptions are included, it's always stated that it couldn't connect. Is there anything that's required within the setup? I am using docker-compose with a nginx and normal websites work properly like the editor itself.

I don't get it, can you share a video of it?

> How do I use the ssh keys added within the machine? It works on the terminal of VS Code but I cannot push any changes within the git UI of VS Code.

I haven't tried that since I'm on Windows but I think @jsjoeio could answer.

**Reply by philippneugebauer** [reply]
> > I set up my code server with docker but I am a bit confused how I am supposed to install the binary for erlang + elixir.
> 
> code-server's base OS image is Debian. I don't know erlang and elixir, but have you tried getting curl, wget and some extra packages and try installing it classically on Linux?

I was just wondering if there's some special stuff to consider. If I just need to install that, then I have done that hopefully now.

> > If some external content like extension descriptions are included, it's always stated that it couldn't connect. Is there anything that's required within the setup? I am using docker-compose with a nginx and normal websites work properly like the editor itself.
> 
> I don't get it, can you share a video of it?

My bad, I misinterpreted the error. X-Frame-Options are to blame. Sorry.

**Reply by jsjoeio** [reply]
> I set up my code server with docker but I am a bit confused how I am supposed to install the binary for erlang + elixir.
> Just install it within the container?

I assume most people extend the image so I'd do that. There are some examples [here](https://github.com/coder/deploy-code-server).

> And another small question. If some external content like extension descriptions are included, it's always stated that it couldn't connect. 

Sounds like you figured this out?

> Is there anything that's required within the setup? I am using docker-compose with a nginx and normal websites work properly like the editor itself.

Do you have something specific in mind? Otherwise, we have some nginx docs [here](https://coder.com/docs/code-server/latest/guide#using-lets-encrypt-with-nginx).

> How do I use the ssh keys added within the machine? It works on the terminal of VS Code but I cannot push any changes within the git UI of VS Code.

Hmm you probably haven't authenticated correctly or your credentials aren't being saved maybe? I'm not sure on that one. You should be able to trigger VS Code to ask for your creds by running one of the Git/GitHub commands from the command palette.

Hope that helps!

**Reply by philippneugebauer** [reply]
> > And another small question. If some external content like extension descriptions are included, it's always stated that it couldn't connect.
> 
> Sounds like you figured this out?

yeah

> > Is there anything that's required within the setup? I am using docker-compose with a nginx and normal websites work properly like the editor itself.
> 
> Do you have something specific in mind? Otherwise, we have some nginx docs [here](https://coder.com/docs/code-server/latest/guide#using-lets-encrypt-with-nginx).

yeah, like the x-frame-option settings and stuff, which seem to cause my trouble

> > How do I use the ssh keys added within the machine? It works on the terminal of VS Code but I cannot push any changes within the git UI of VS Code.
> 
> Hmm you probably haven't authenticated correctly or your credentials aren't being saved maybe? I'm not sure on that one. You should be able to trigger VS Code to ask for your creds by running one of the Git/GitHub commands from the command palette.
> 
> Hope that helps!

Where can I enter them? I wasn't able to find sth to put my ssh keys in so I assumed the ssh folder was enough but it doesn't seem to be

**Reply by jsjoeio** [reply]
> Where can I enter them? I wasn't able to find sth to put my ssh keys in so I assumed the ssh folder was enough but it doesn't seem to be

Hmm...I've only used a PAT to publish to GitHub 🤔 (not ssh keys). 

## Video

https://user-images.githubusercontent.com/3806031/153943607-76949f82-dc1e-4a4a-869d-bb53140c47e9.mov


How would you do it in a local dev env? 

Maybe @bpmct knows
